hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/util/StreamerUtil.java

```diff
@@ -478,4 +481,36 @@ public static boolean isWriteCommit(HoodieTableType tableType, HoodieInstant ins
   public static String getAuxiliaryPath(Configuration conf) {
     return conf.getString(FlinkOptions.PATH) + Path.SEPARATOR + AUXILIARYFOLDER_NAME;
   }
+
+  /**
+   * Check the {@link FlinkOptions#PRECOMBINE_FIELD} of configuration.
+   *
+   * @param conf   The flink configuration.
+   * @param schema The table schema.
+   */
+  public static void checkPreCombineField(Configuration conf, ResolvedSchema schema) {
+    checkPreCombineField(conf, schema.getColumnNames());
+  }
+
+  /**
+   * Check the {@link FlinkOptions#PRECOMBINE_FIELD} of configuration.
+   *
+   * @param conf        The flink configuration.
+   * @param columnNames The column names.
+   */
+  public static void checkPreCombineField(Configuration conf, List<String> columnNames) {
+    String preCombineField = conf.get(FlinkOptions.PRECOMBINE_FIELD);
+    if (!columnNames.contains(preCombineField)) {
```

Comment:
Does it still throw after spark pkless support: https://github.com/apache/hudi/pull/8107

Reply:
@danny0405, it should still throw after spark pkless support #8107, because #8107 is used to auto generation of record keys not precombine field.

Reply:
After #8107, spark will not throw exception if there is no primary key, maybe we should just fix the primary key set up.

Reply:
@SteNicholas This comment does not look like resolved. Can you please revisit?